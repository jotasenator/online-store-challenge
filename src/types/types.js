export const types = {
    get: 'GET-DATA',
    error: 'ERROR',
    selected: 'SELECTED-DATA',
    delete: 'DELETED-DATA',
    modify: 'MODIFIED-DATA'
}
