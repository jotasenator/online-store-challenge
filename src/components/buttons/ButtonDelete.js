import React from 'react'
import { getItemDeleteGetItems } from '../../actions/actions'
import { useDispatch } from 'react-redux'

export default function ButtonDelete({ id }) {

    const dispatch = useDispatch()

    const handleDelete = () => {
        dispatch(getItemDeleteGetItems(id))
    }

    return (
        <div>
            <button
                className='btn btn-danger'
                onClick={handleDelete}
            >
                Delete
            </button>
        </div>
    )
}
