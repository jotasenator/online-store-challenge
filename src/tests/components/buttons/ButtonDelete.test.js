import React from 'react'
import '@testing-library/jest-dom'
import { mount } from 'enzyme'

import thunk from "redux-thunk";
import { Provider } from "react-redux"
import configureStore from "redux-mock-store";

import ButtonDelete from '../../../components/buttons/ButtonDelete'
import { types } from '../../../redux/types/types';


describe('Test from ButtonDelete', () => {
    const id = 10
    const middlewares = [thunk]
    const mockStore = configureStore(middlewares)
    const initState = {
        data: null,
        selected: null,
        deleted: '',
        created: null,
        modified: null,
        error: ''
    };
    const store = mockStore(initState);

    const wrapper = mount(
        <Provider store={store}>
            <ButtonDelete id={id} />
        </Provider>
    )

    const getData = () => ({ type: 'GET-DATA' })

    // Dispatch the action
    store.dispatch(getData())


    test('should render <ButtonDelete/> correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })
    test('should show text inside button tag', () => {
        expect((wrapper).find('button').text()).toBe('Delete')
        expect((wrapper).find('button[className="btn btn-danger"]').text()).toBe('Delete')
    })
    test('should store dispatch expected action', () => {
        // Test if your store dispatched the expected actions
        const actions = store.getActions()
        const expectedPayload = { type: types.get }
        expect(actions).toEqual([expectedPayload])
    })
})

