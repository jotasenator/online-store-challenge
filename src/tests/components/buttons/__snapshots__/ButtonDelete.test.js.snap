// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test from ButtonDelete should render <ButtonDelete/> correctly 1`] = `
<div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
